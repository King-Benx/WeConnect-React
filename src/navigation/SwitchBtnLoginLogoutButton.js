import React from "react";
import CustomFunctions from "../custom/CustomFunctions";
import Login from "../navigation/Login";
import Search from "../containers/business/Search";
import Logout from "../navigation/Logout";
class SwitchBtnLoginLogoutButton extends React.Component {
  render() {
    return CustomFunctions.checkAuthentication() ? (
      // JSX returned switches between login and logout
      <ul className="nav navbar-nav navbar-right">
        <Search />
        <Logout />
      </ul>
    ) : (
      <ul className="nav navbar-nav navbar-right">
        <Login />
      </ul>
    );
  }
}
export default SwitchBtnLoginLogoutButton;
