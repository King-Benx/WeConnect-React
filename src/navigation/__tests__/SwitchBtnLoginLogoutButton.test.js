import React from 'react';
import { shallow } from 'enzyme';
import SwitchBtnLoginLogoutButton from "../SwitchBtnLoginLogoutButton";
import renderer from 'react-test-renderer';

describe('These are tests the logout link', () => {
    const event={
        target:{
            value:{}
        },
        preventDefault: () => {

        }
    }
    
    let mountedSwitch;

    beforeEach(()=>{
        mountedSwitch = shallow(<SwitchBtnLoginLogoutButton />);
    });

    it('renders without crashing', () => {
        shallow(<SwitchBtnLoginLogoutButton />)
        const tree = renderer.create(<SwitchBtnLoginLogoutButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has a ul tag', () =>{
        const uls = mountedSwitch.find('ul');
        expect(uls.length).toBe(1)
    });

    it('has a Login', () =>{
        const login = mountedSwitch.find('Login');
        expect(login.length).toBe(1)
    });

});