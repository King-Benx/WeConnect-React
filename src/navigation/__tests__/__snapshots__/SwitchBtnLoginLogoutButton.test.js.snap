// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`These are tests the logout link renders without crashing 1`] = `
<ul
  className="nav navbar-nav navbar-right"
>
  <li>
    <a
      href="/login"
      onClick={[Function]}
    >
      <b>
        <i
          className="glyphicon glyphicon-log-in"
        />
         Login
      </b>
    </a>
  </li>
</ul>
`;
